client/add/box.ctrl.js

```diff
@@ -0,0 +1,22 @@
+const Constants = require('../../api/services/Constants');
+/**
+ * Controller for handling the dialog to add a new box
+ * @return {function} A controller that contains 2 test elements
+ */
+module.exports = function ($mdDialog) {
+  this.name = '';
+  this.description = '';
+  this.hide = function() {
+    $mdDialog.hide();
+  };
+  this.cancel = function() {
+    $mdDialog.cancel();
+  };
+  this.answer = function() {
+    $mdDialog.hide({
+      name: this.name,
+      description: this.description,
+      visibility: Constants.DEFAULT_BOX_VISIBILITY_SETTING
```

Comment:
Ideally this should probably be the user's `defaultVisibilitySetting` preference -- `Constants.DEFAULT_BOX_VISIBILITY_SETTING` is the default value of that preference for all users, but if the current user change their preference then this line will use the wrong setting.